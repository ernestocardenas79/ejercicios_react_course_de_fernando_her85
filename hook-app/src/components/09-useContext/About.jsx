import { useContext } from 'react';
import { UserContext } from './UserContext';

export const About = () => {
  const {user, setUser} = useContext(UserContext);

  return (
    <div>
      <h1>About</h1>
      <hr/>

      <pre>
        {JSON.stringify(user,null, 4)}
      </pre>

      <button className="btn btn-warning" onClick={()=>setUser({})}>Logout</button>
    </div>
  );
};
