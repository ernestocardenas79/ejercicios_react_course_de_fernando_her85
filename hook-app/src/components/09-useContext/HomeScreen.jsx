import { UserContext } from './UserContext';
import { useContext } from 'react';

export const HomeScreen = () => {
  const {user} = useContext(UserContext);

  return (
    <div>
      <h1>HomeScreen</h1>
      <hr />
      <pre aria-label='pre'>
        { JSON.stringify(user, null, 4)}
      </pre>
    </div>
  );
};
