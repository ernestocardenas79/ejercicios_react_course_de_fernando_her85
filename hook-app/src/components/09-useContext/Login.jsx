import { useContext } from 'react';
import { UserContext } from './UserContext';

export const Login = () => {
  const {setUser} = useContext(UserContext);
  const user = {
    id:774,
    name:'Ernesto'
  }
  return (
    <div>
      <h1>Login</h1>
      <button className="btn btn-primary" onClick={()=>setUser(user)}>
        Login
      </button>
    </div>
  );
};
