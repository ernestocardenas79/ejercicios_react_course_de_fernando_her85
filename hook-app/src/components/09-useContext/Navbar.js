import React from "react";
import { NavLink } from "react-router-dom";

export const Navbar = () => {
  return (
    <nav class="navbar navbar-expand-sm navbar-dark bg-dark">
      <h1>useContext</h1>

      <div className="collapse navbar-collapse" id="navbarNav">
        <ul className="navbar-nav">
          <li className="nav-item">
            <NavLink
              exact
              activeClassName="active"
              className="nav-link"
              to="./"
            >
              Home <span class="sr-only">(current)</span>
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink
              exact
              activeClassName="active"
              className="nav-link"
              to="./about"
            >
              About
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink
              exact
              activeClassName="active"
              className="nav-link"
              to="./login"
            >
              Login
            </NavLink>
          </li>
        </ul>
      </div>
    </nav>
  );
};
