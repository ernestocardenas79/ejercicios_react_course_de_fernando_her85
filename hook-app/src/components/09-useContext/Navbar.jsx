import { NavLink, Link } from 'react-router-dom';

export const Navbar = () => {
  return (
    <nav className='navbar navbar-expand-sm navbar-dark bg-dark rounded-3'>
      <Link className="navbar-brand" to="/">useContext</Link>

      <div className='collapse navbar-collapse' id='navbarNav'>
        <ul className='navbar-nav'>
          <li className='nav-item'>
            <NavLink
              className={({isActive})=>`nav-link ${ isActive? 'active':''}`}
              to='/'
            >
              Home <span className='sr-only'>(current)</span>
            </NavLink>
          </li>
          <li className='nav-item'>
            <NavLink
               className={({isActive})=>`nav-link ${ isActive? 'active':''}`}
              to='about'
            >
              About
            </NavLink>
          </li>
          <li className='nav-item'>
            <NavLink
              className={({isActive})=>`nav-link ${ isActive? 'active':''}`}
              to='login'
            >
              Login
            </NavLink>
          </li>
        </ul>
      </div>
    </nav>
  );
};
