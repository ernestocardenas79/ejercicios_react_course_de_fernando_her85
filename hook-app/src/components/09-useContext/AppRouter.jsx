import { About } from './About';
import { Login } from './Login';
import { HomeScreen } from './HomeScreen';
import {
  Route,
  Navigate,
  Routes,
} from 'react-router-dom';

export const AppRouter = () => {
  return (
    <Routes>
      <Route  path='/' element={<HomeScreen/>} />
      <Route  path='about' element={<About/>} />
      <Route  path='login' element={<Login/>} />
      <Route  path='/*' element={<Navigate to="/login"/>} />
    </Routes>
  );
};
