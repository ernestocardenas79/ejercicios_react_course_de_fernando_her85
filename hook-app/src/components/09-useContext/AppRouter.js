import React from 'react';
import { About } from './About';
import { Login } from './Login';
import { HomeScreen } from './HomeScreen';
import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom';
import { Navbar } from './Navbar';

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <Navbar />
        <div className="container">
        <Switch>
          <Route exact path='/about' component={About} />
          <Route exact path='/login' component={Login} />
          <Route exact path='/' component={HomeScreen} />
          <Redirect component={HomeScreen} />
        </Switch>
        </div>
      </div>
    </Router>
  );
};
