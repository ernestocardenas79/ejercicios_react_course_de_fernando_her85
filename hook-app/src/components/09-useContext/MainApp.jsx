import { Navigate, Route, Routes } from 'react-router-dom';
import { UserContext } from './UserContext';
import { useState } from 'react';
import { HomeScreen } from './HomeScreen';
import { About } from './About';
import { Login } from './Login';
import { Navbar } from './Navbar';

import './styles.css'

export const MainApp = () => {
 const [user, setUser] = useState({});

  return (
    <UserContext.Provider value={{user, setUser}}>
      <Navbar />
      <hr />
              
      <Routes>
          <Route  path='/' element={<HomeScreen/>} />
          <Route  path='about' element={<About/>} />
          <Route  path='login' element={<Login/>} />
          <Route  path='/*' element={<Navigate to="/login"/>} />
      </Routes>   
    </UserContext.Provider>
  );
};


   