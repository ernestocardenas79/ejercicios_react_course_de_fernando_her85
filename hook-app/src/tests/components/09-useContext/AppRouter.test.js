import { UserContext } from '../../../components/09-useContext/UserContext';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

describe('Pruebas en <AppRoute>', () => {
  const user = {
    id: 1,
    name: 'Ernesto',
  };


  test('Debe de mostrarse correctamente', () => {
    render(
      <MemoryRouter>
      <UserContext.Provider value={{ user }}>
        <AppRouter />
      </UserContext.Provider>
      </MemoryRouter>
    );

    expect(screen.getByText('HomeScreen')).toBeTruthy();
  });

  test('Debe de mostrar el LoginPage', () => {
    render(
      <MemoryRouter initialEntries={['/login']}>
      <UserContext.Provider value={{ user }}>
        <AppRouter />
      </UserContext.Provider>
      </MemoryRouter>
    );
    expect(screen.getByLabelText('h1').innerHTML).toBe('Login');
  });
});
