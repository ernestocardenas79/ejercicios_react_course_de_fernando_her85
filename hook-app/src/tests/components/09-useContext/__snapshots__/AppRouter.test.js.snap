// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRoute> Debe de mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <h1>
      useContext
    </h1>
    <div
      className="collapse navbar-collapse"
      id="navbarNav"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home 
            <span
              className="sr-only"
            >
              (current)
            </span>
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
    "id": 1,
    "name": "Ernesto"
}
      </pre>
    </div>
  </div>
</div>
`;
